File: foxy/parent/src/command/vanilla/util/DblExecutor.ts
import { createEmbed } from "../../../utils/discord/Embed";
import { ExecutorParams } from "../../structures/CommandExecutor";

export default class DblExecutor {
    async execute({ context, endCommand, t }: ExecutorParams) {
        const embed = createEmbed({
            description: t('commands:upvote.description')
        });

        context.reply({ embeds: [embed] });

        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/discord/DblExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.discord

import net.cakeyfox.common.Colors
import net.cakeyfox.common.Constants
import net.cakeyfox.common.FoxyEmotes
import net.cakeyfox.foxy.interactions.commands.CommandContext
import net.cakeyfox.foxy.interactions.commands.UnleashedCommandExecutor
import net.cakeyfox.foxy.utils.linkButton

class DblExecutor : UnleashedCommandExecutor() {
    override suspend fun execute(context: CommandContext) {
        context.reply {
            embed {
                description = context.locale["dbl.embed.description"]
                color = Colors.BLURPLE
            }

            actionRow(
                linkButton(
                    FoxyEmotes.FoxyYay,
                    context.locale["dbl.clickHereToVote"],
                    Constants.UPVOTE_URL
                )
            )
        }
    }
}